geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java

```diff
@@ -555,7 +556,8 @@ boolean doTCPCheckMember(InternalDistributedMember suspectMember, int port) {
           // expected
         }
       }
-    } while (!passed && !this.isShutdown() && System.nanoTime() < giveupTime);
+    } while (retryIfConnectFails && !passed && !this.isShutdown()
```

Comment:
From the PR title it sounds like the issue was that we aren't pinging the suspect member enough times before giving up (immediately initiating removal).  Why doesn't the giveupTime handle that problem?  We should continue to retry for up to the configured member timeout.  Since we are &&'ing all these conditions together, won't it still break if the timeout is exceeded even if retryIfConnectFails is true?  Forgive me if these are amateur questions, not super familiar with this code.